idr/query.go

```diff
@@ -0,0 +1,96 @@
+package idr
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/antchfx/xpath"
```

Comment:
Just realising that we are still using antchfx's xpath, so will you move away it in the future, or we will still use its query functionalities as it is common?

Reply:
Xpath absolutely stays. Never had any intention of moving away from it. It serves as well and it is still internal to parser. The reason for back and forth on  Node is:

We did our node originally. But it was direct copy of XML node and streaming parsing had severe limitations. So I thought I should contribute back to antchfx/xmlquery so we can just use it's Node plus implement a nice streaming parser in it.

After that was done the more I looked at xmlquery Node the unhappier I became due to the fact that xmlquery.Node is for XML and if we eventually make ingested.Read return Node it would be weird to return XML Node.

So here we come IDR.

Its xpath is completely good for us.

Reply:
Got it